### ISSUE TITLE
Implement Navigation

### ISSUE THREAD

**Issue by FilipMessa** [kiwicom/margarita#37]
- NextJS provide navigation between pages out of that box, but we need to solve this also for react-native.

**Comment by tbergq** [comment]
I think react-navigation is the only choice when it comes to expo-apps. 

I also think it is time to use the workspaces, and have a structure like

apps/
----- web/
----- mobile/
----- shared(core) /

with web containing nextjs pages, which uses components from shared app and 
mobile containing react-navigation which also uses components from shared.

what to you think?
@FilipMessa @RobinCsl

**Comment by RobinCsl** [comment]
Ah, so you would like to decouple entirely the navigation and have it per platform?
I just hope we can find a satisfactory way to reuse logic, because there are some key differences (passing data as params in `react-navigation` and using url params in next.js?)

**Comment by tbergq** [comment]
Yeah, but these differences are there even if we keep it all together right?

**Comment by RobinCsl** [comment]
I guess you're right. 
So we want to roll with `react-navigation` for mobile, and next.js's default navigation for web. Agreed?
So we might need some kind of wrapper to abstract these away? 
I will try and build the navigation of the screens we need and see what problems I encounter.

**Comment by FilipMessa** [comment]
From me it sounds like a nice way, I don't have any better idea so let's try.
BTW: I would like to also split graphql from the app as I mention here https://skypicker.slack.com/archives/CEFT2NAE8/p1544279888003100.

**Comment by tbergq** [comment]
Yes, I saw it, it could also go into app folder, 
so we have apps -> graphql, web, native, core and packages that supports the apps

**Comment by tbergq** [comment]
We don't have navigation yet? Or did we just forget to close this issue?
